soroban-env-host/src/host.rs

```diff
@@ -2377,6 +2384,67 @@ impl VmCallerEnv for Host {
         self.create_contract_with_id_preimage(code, id_preimage)
     }
 
+    // Notes on metering: covered by components
+    fn put_tmp_contract_data(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        k: RawVal,
+        v: RawVal,
+    ) -> Result<Void, HostError> {
+        let key = self.from_host_val(k)?;
+        self.0.temp_storage.borrow_mut().put(
+            self.get_current_contract_id_internal()?.0,
```

Comment:
I think before this PR we should implement https://github.com/stellar/rs-soroban-env/issues/681 so that we have a cached `RawVal` on hand we can just reuse as the current contract ID.

Reply:
I don't think this should depend on such optimizations - with rc the key sizes will be sufficiently small and this anyway is a very minor optimization comparing to the regular storage that stores entire ledger keys.

Reply:
*Shrug* ok I would like to do it eventually but if people are blocked on this we can land now.